sdk/search/azure-search/src/main/java/com/azure/search/SearchIndexClientBuilder.java

```diff
@@ -34,81 +33,131 @@
  * SearchIndexClients} and {@link SearchIndexAsyncClient SearchIndexAsyncClients}. Call {@link #buildClient()
  * buildClient} and {@link #buildAsyncClient() buildAsyncClient} respectively to construct an instance of the desired
  * client.
- *
- * The following information must be provided on this builder: the Azure Cognitive Search service endpoint through
- * {@code .endpoint()} the index name through {@code .indexName()} the API key through {@code .credential()}
+ * <p>
+ * The following information must be provided to successfully create a client.
+ * <ul>
+ *     <li>{@link #endpoint(String)}</li>
+ *     <li>{@link #indexName(String)}</li>
+ *     <li>{@link #credential(SearchApiKeyCredential)} or {@link #pipeline(HttpPipeline)}</li>
+ * </ul>
  */
 @ServiceClientBuilder(serviceClients = {SearchIndexClient.class, SearchIndexAsyncClient.class})
 public class SearchIndexClientBuilder {
-
-    // This header tells the server to return the request id in the HTTP response. Useful for correlation with what
-    // request was sent.
+    /*
+     * This header tells the service to return the request ID in the HTTP response. This is useful for correlating the
+     * request sent to the response.
+     */
     private static final String ECHO_REQUEST_ID_HEADER = "x-ms-return-client-request-id";
 
+    /*
+     * This is the name of the request ID header that the Search service expects. If the customer doesn't pass
+     * RequestOptions into a service call this will allow for the client to implicitly add a request ID header,
+     * otherwise the passed request ID will be used.
+     */
+    private static final String CLIENT_REQUEST_ID = "client-request-id";
+
     private static final String SEARCH_PROPERTIES = "azure-search.properties";
     private static final String NAME = "name";
     private static final String VERSION = "version";
 
-    SearchApiKeyCredential searchApiKeyCredential;
-    SearchServiceVersion apiVersion;
-    String endpoint;
-    HttpClient httpClient;
-    HttpLogOptions httpLogOptions;
-    Configuration configuration;
-    List<HttpPipelinePolicy> policies;
-    private String clientName;
-    private String clientVersion;
+    private final ClientLogger logger = new ClientLogger(SearchIndexClientBuilder.class);
+    private final List<HttpPipelinePolicy> policies = new ArrayList<>();
+    private final HttpHeaders headers = new HttpHeaders().put(ECHO_REQUEST_ID_HEADER, "true");
 
+    private final String clientName;
+    private final String clientVersion;
+
+    private SearchApiKeyCredential searchApiKeyCredential;
+    private SearchServiceVersion apiVersion;
```

Comment:
I have renamed this field. I am going to check in https://github.com/Azure/azure-sdk-for-java/pull/8393. Please merge my PR.

Reply:
Will do